src/infra/core/tests/tests/ingest/test_fetch.rs

```diff
@@ -686,6 +630,387 @@ async fn test_fetch_container_ok() {
+        assert_eq!(
+            std::fs::read_to_string(&target_path).unwrap(),
+            CSV_BATCH_OUTPUT
+        );
+
+        match res_2 {
+            FetchResult::Updated(x) => x.source_state,
+            _ => unreachable!(),
+        }
+    };
+    let prev_source_state = {
+        let fetch_step_3 = FetchStep::Container(FetchStepContainer {
+            image: BUSYBOX.to_owned(),
+            command: Some(vec!["sh".to_owned()]),
+            args: Some(vec!["-c".to_owned(), HAS_MORE_TESTER_SCRIPT.to_owned()]),
+            env: Some(vec![EnvVar {
+                name: ODF_BATCH_SIZE.to_owned(),
+                value: Some(custom_batch_size.to_string()),
+            }]),
+        });
+
+        let res_3 = fetch_svc
+            .fetch(
+                "3",
+                &fetch_step_3,
+                prev_source_state.as_ref(),
+                &target_path,
+                &Utc::now(),
+                None,
+            )
+            .await
+            .unwrap();
+
+        assert_matches!(
+            &res_3,
+            FetchResult::Updated(FetchResultUpdated {
+                source_state: Some(PollingSourceState::ETag(expected_etag)),
+                has_more: false,
+                ..
+            }) if expected_etag == "100"
+        );
+        assert!(target_path.exists());
+        assert_eq!(
+            std::fs::read_to_string(&target_path).unwrap(),
+            CSV_BATCH_OUTPUT
+        );
+
+        match res_3 {
+            FetchResult::Updated(x) => x.source_state,
+            _ => unreachable!(),
+        }
+    };
+
+    {
+        let fetch_step_4 = FetchStep::Container(FetchStepContainer {
+            image: BUSYBOX.to_owned(),
+            command: Some(vec!["sh".to_owned()]),
+            args: Some(vec!["-c".to_owned(), HAS_MORE_TESTER_SCRIPT.to_owned()]),
+            env: Some(vec![EnvVar {
+                name: ODF_BATCH_SIZE.to_owned(),
+                value: Some(custom_batch_size.to_string()),
+            }]),
+        });
+
+        let res_4 = fetch_svc
+            .fetch(
+                "4",
+                &fetch_step_4,
+                prev_source_state.as_ref(),
+                &target_path,
+                &Utc::now(),
+                None,
+            )
+            .await
+            .unwrap();
+
+        assert_matches!(res_4, FetchResult::UpToDate);
+        assert!(target_path.exists());
+        assert_eq!(std::fs::read_to_string(&target_path).unwrap(), "");
+    }
+}
+
+///////////////////////////////////////////////////////////////////////////////
+// Utils: constants
+///////////////////////////////////////////////////////////////////////////////
+
+const CSV_BATCH_OUTPUT: &str = indoc!(
+    "
+    city,population
+    A,1000
+    B,2000
+    C,3000
+    "
+);
+const HAS_MORE_TESTER_SCRIPT: &str = indoc! {r#"
+    #!/usr/bin/env sh
+
+    set -euo pipefail
+
+    if [[ -n "${DEBUG:-}" ]]; then
+      set -x
+    fi
+
+    ROWS_COUNT=100
```

Comment:
I found it hard to understand the tests without closely reading this script, and it's mainly because it **hardcodes** the row count at 100.

I feel like it would be a bit simpler to understand if `ROW_COUNT` was passed as a parameter, so that in tests you could clearly see how `ROW_COUNT` compares to `ODF_BATCH_SIZE`.

Reply:
It definitely makes sense -- Done